templates/vue/src/components/lightbox/quiz/Question.vue

```diff
@@ -0,0 +1,174 @@
+<template>
+  <div class="question">
+    <div v-show="formOpened" ref="form-container"></div>
+    <div v-show="!formOpened">
+      <h1 class="question-title">
+        <div class="question-title-step">
+          {{ currentStep }}
+        </div>
+        {{ question.text }}
+        <div
+          class="question-title-bubble"
+          :style="{ backgroundImage: bubbleImage }"
+        ></div>
+      </h1>
+      <div class="question-content">
+        <p class="question-answer-text">I want to answer with...</p>
+        <div class="button-container">
+          <answer-button :disabled="disableButton" @click="openForm(question.answers.textId)">text</answer-button>
+          <answer-button :disabled="disableButton" icon="microphone">audio</answer-button>
+          <answer-button :disabled="disableButton" icon="tasks" @click="openForm(question.answers.checklistId)">checklist</answer-button>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import AnswerButton from "./AnswerButton"
+import TapestryAPI from "../../../services/TapestryAPI"
+import SpeechBubble from "../../../assets/speech-bubble-end.png"
+
+export default {
+  name: "question",
+  components: {
+    AnswerButton,
+  },
+  props: {
+    question: {
+      type: Object,
+      required: true,
+    },
+    currentStep: {
+      type: String,
+      required: false,
+      default: "1/1",
+    },
+  },
+  data() {
+    return {
+      formOpened: false,
+      disableButton: false,
+    }
+  },
+  computed: {
+    bubbleImage() {
+      return `url(${wpData.vue_uri}/${SpeechBubble.split("dist")[1]})`
+    },
+  },
+  methods: {
+    async openForm(id) {
+      if (!id) {
+        return
+      }
+
+      // Clear previous form data
+      delete window[`gf_submitting_${id}`]
+      this.$refs["form-container"].innerHTML = ""
+      this.disableButton = true;
+
+      const TapestryApi = new TapestryAPI(wpPostId)
+      try {
+        const response = await TapestryApi.getGravityForm(id)
+        if (response) {
+          const gravityForm = document.createElement("div")
+          gravityForm.innerHTML = response.data
+          gravityForm
+            .querySelector(`form#gform_${id}`)
+            .addEventListener("submit", (event) => {
+              this.formOpened = false
+              this.$emit("form-submitted")
+              this.disableButton = false;
+            })
+
+          this.$refs["form-container"].appendChild(gravityForm)
+          this.formOpened = true
+          this.$emit("form-opened")
```

Comment:
We should avoid doing any direct manipulation of the DOM. Vue has a `v-html` attribute that you can use to inject html, and events "bubble" so you can attach the submit handler to the `form-container` element and it should work fine.

Reply:
Please feel free to modify the code. Thanks.

Reply:
Yes, I agree with Narendra here. We should avoid doing things through plain JS when there is a vue way of doing it. Please fix this.